plugins/randpa_plugin/include/eosio/randpa_plugin/randpa.hpp

```diff
@@ -451,10 +468,31 @@ class randpa {
         return precommited_keys.size() > node->active_bp_keys.size() * 2 / 3;
     }
 
+    void on(uint32_t ses_id, const finality_notice_msg& msg) {
+        const auto& data = msg.data;
+        dlog("Randpa finality_notice_msg received for block ${bid}", ("bid", data.best_block));
+        if (is_active_bp(data.best_block) && get_block_num(data.best_block) == _last_prooved_block_num) {
```

Comment:
I think `get_block_num(data.best_block) <= _last_prooved_block_num` will better, because cover all cases, e.g. when we handle best_block from "future" while sync